velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp

```diff
@@ -58,6 +58,79 @@ TEST_F(ParquetReaderTest, parseSample) {
   EXPECT_EQ(type->childByName("b"), col1);
 }
 
+TEST_F(ParquetReaderTest, parseReadAsLowerCase) {
+  // upper.parquet holds two columns (A: BIGINT, b: BIGINT) and
+  // 2 rows.
+  const std::string upper(getExampleFilePath("upper.parquet"));
+
+  ReaderOptions readerOptions{defaultPool.get()};
+  readerOptions.setReadAsLowerCase(true);
+  ParquetReader reader = createReader(upper, readerOptions);
+  EXPECT_EQ(reader.numberOfRows(), 2ULL);
+
+  auto type = reader.typeWithId();
+  EXPECT_EQ(type->size(), 2ULL);
+  auto col0 = type->childAt(0);
+  EXPECT_EQ(col0->type->kind(), TypeKind::BIGINT);
+  auto col1 = type->childAt(1);
+  EXPECT_EQ(col1->type->kind(), TypeKind::BIGINT);
+  EXPECT_EQ(type->childByName("a"), col0);
+  EXPECT_EQ(type->childByName("b"), col1);
+}
+
+TEST_F(ParquetReaderTest, parseRowMapArrayReadAsLowerCase) {
+  // upper_complex.parquet holds one row of type
+  // root
+  //  |-- Cc: struct (nullable = true)
+  //  |    |-- CcLong0: long (nullable = true)
+  //  |    |-- CcMap1: map (nullable = true)
+  //  |    |    |-- key: string
+  //  |    |    |-- value: struct (valueContainsNull = true)
```

Comment:
Move the struct inside the array

Reply:
Do you mean the array element is also struct?